.github/workflows/ci-tests.yml

```diff
@@ -92,6 +92,11 @@ jobs:
           restore-keys: |
             ${{ runner.os }}-go-
 
+      - name: Run CI benchmarks
+        id: ci-benchmarks
+        run: |
+          ./bin/run-benchmarks.sh 2>&1
```

Comment:
This will add up to the CI time. I'd suggest splitting it off to an improvements PR so we create a new job and update the pipeline a bit.